### PR TITLE
1661 contact info

### LINKED ISSUE #1661: Rework patient contact information
Motivated by #1423 we want a more generic way to store patient relations. Currently, there is a family object with predefined entries like `mother` or `nextOfKin`. For things like #1423 we want an automatic way to extract this information though. Furthermore, currently it's not possible to have, for example, multiple `nextOfKin`.

A solution would be to have a contact array of `RelatedPerson` objects instead.

Currently, we display an object from the family object. This wouldn't be possible with a contact array, and we need to have a new control to pick and display certain contacts from the contact array.

### PR DESCRIPTION
**Pull request by clemens-msupply**
Fixes #1661

This PR goes together with the `1661-contact-info` branch in the schema repository and the latest develop branch for mSupply.

# 👩🏻‍💻 What does this PR do? 
- Removes the family object from the patient schema
- Adds a `contacts` array to the patient schema
- Adds a `contacts` array to the enrolment schema
- Adds a new generic JSONForms control which is used to display and edit an entry from the contacts array
- Add a search component to lookup an entry from the patient contacts

The RelatedPerson has a `category` field now. This can, for example, be used to mark a contact as "Mother" or "Next of Kin". The "Next of Kin" can then be picked from the contacts list using the category field.
However, there is still the `relationship` field which can be used, for example, like:
```ts
{
  category: 'NextOfKin',
  relationship: 'Father',
}
```

# 🧪 How has/should this change been tested? 

- Create/ update a new patient and edit the next of kin contact
- In the HIV care enrolment select "same as next of kin" for the treatment supporter and check it load correctly

### CODE REVIEW

**Review by roxy-dao — APPROVED** [review]
LGTM~ Works great

File: client/packages/programs/src/JsonForms/common/components/Array/KeyedItemArray.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  const { errors: zErrors, options } = useZodOptionsValidation(
+    Options,
+    uischema.options
+  );
+
+  useEffect(() => {
+    if (!options) return;
+    const arrayData = data ?? [];
+    const elementIndex = arrayData.findIndex(it => {
+      if (!it || typeof it !== 'object' || Array.isArray(it)) return;
```
Comment (roxy-dao) [inline client/packages/programs/src/JsonForms/common/components/Array/KeyedItemArray.tsx:72]: nit: can `it`s be renamed?

File: client/packages/programs/src/JsonForms/components/Search/Search.tsx [hunk 2]
```diff
@@ -48,10 +49,29 @@ const DocumentOptions = z.object({
   saveFields: z.array(z.string()).optional(),
 });
 
-const Options = z.discriminatedUnion('source', [DocumentOptions, UserOptions]);
+const PatientContactOptions = z.object({
+  /**
+   * Source of the search data -- looks up the first contact from the patient contact list using
+   * with has the matching `category`.
+   */
```
Comment (roxy-dao) [inline client/packages/programs/src/JsonForms/components/Search/Search.tsx:56]:
```suggestion
   * Source of the search data -- looks up the first contact from the patient contact list using
   * the matching `category`.
   */
```

File: client/packages/programs/src/JsonForms/common/components/Array/KeyedItemArray.tsx (hunk 1 above)
Comment (clemens-msupply) [reply client/packages/programs/src/JsonForms/common/components/Array/KeyedItemArray.tsx:72]: Got this from Kotlin, they have an implicit `it` parameter in lambdas... Can call it `item`. Like `it` though, `item` or `element` are IMHO not much better but longer...

**Pull request merged**